xline-client/src/types/watch.rs

```diff
@@ -142,55 +144,63 @@ impl WatchRequest {
         self
     }
 
-    /// Set `range_end`
+    /// `range_end` is the end of the range [key, `range_end`) to watch. If `range_end` is not given,
+    /// only the key argument is watched. If `range_end` is equal to '\0', all keys greater than
+    /// or equal to the key argument are watched.
+    /// If the `range_end` is one bit larger than the given key,
+    /// then all keys with the prefix (the given key) will be watched.
     #[inline]
     #[must_use]
     pub fn with_range_end(mut self, range_end: impl Into<Vec<u8>>) -> Self {
         self.inner.range_end = range_end.into();
         self
     }
 
-    /// Set `start_revision`
+    /// Sets the start revision to watch from (inclusive). No `start_revision` is "now".
     #[inline]
     #[must_use]
     pub const fn with_start_revision(mut self, revision: i64) -> Self {
         self.inner.start_revision = revision;
         self
     }
 
-    /// Set `progress_notify`
+    /// `progress_notify` is set so that the Xline server will periodically send a `WatchResponse` with no events to the new watcher if there are no recent events. It is useful when clients wish to recover a disconnected watcher starting from a recent known revision. The xline server may decide how often it will send notifications based on current load.
     #[inline]
     #[must_use]
     pub const fn with_progress_notify(mut self) -> Self {
         self.inner.progress_notify = true;
         self
     }
 
-    /// Set `filters`
+    /// filters filter the events at server side before it sends back to the watcher.
     #[inline]
     #[must_use]
     pub fn with_filters(mut self, filters: impl Into<Vec<WatchFilterType>>) -> Self {
         self.inner.filters = filters.into().into_iter().map(Into::into).collect();
         self
     }
 
-    /// Set `prev_kv`
+    /// If `prev_kv` is set, created watcher gets the previous KV before the event happens.
+    /// If the previous KV is already compacted, nothing will be returned.
     #[inline]
     #[must_use]
     pub const fn with_prev_kv(mut self) -> Self {
         self.inner.prev_kv = true;
         self
     }
 
-    /// Set `watch_id`
+    /// If `watch_id` is provided and non-zero, it will be assigned to this watcher.
```

Comment:
Non-zero indicates that a user can use a negative number as the watch_id?

Reply:
yes, there's no constraint that the watch_id must be non-negative in both etcd and xline.